### PR TITLE
[receiver/googlecloudmonitoringreceiver] Move googlecloudmonitoringreceiver component stability to alpha

### PR DESCRIPTION
**Pull request by abhishek-at-cloudwerx**
**Description:** This PR moves the stability of the googlecloudmonitoringreceiver component to alpha

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33762

**Testing:** No additional test cases were added

**Documentation:** Previously updated

### CODE REVIEW

**Review by dashpole — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
needs a `make gotidy`

File: cmd/otelcontribcol/builder-config.yaml [hunk 1]
```diff
@@ -125,6 +125,7 @@ processors:
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor v0.109.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor v0.109.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/remotetapprocessor v0.109.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver v0.109.0
```
Comment (codeboten) [inline cmd/otelcontribcol/builder-config.yaml:128]: Submitted a [fix](https://github.com/Internal-Cloudwerx/opentelemetry-collector-contrib/pull/1), the receiver is in the wrong group in the config

**Review by TylerHelmuth — APPROVED** [review]

**Comment by datsabk** [comment]
@TylerHelmuth - Could you please proceed with the merge

**Comment by codeboten** [comment]
@datsabk it looks like there's still 16 failures in the PR, those will need to get resolved before it's merged

**Comment by dashpole** [comment]
Needs:

* `make gotidy`
* `make genotelcontribcol`

File: cmd/otelcontribcol/go.mod [hunk 2]
```diff
@@ -2,9 +2,7 @@
 
 module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontribcol
 
-go 1.22.0
-
-toolchain go1.22.7
+go 1.22.7
```
Comment (codeboten) [inline cmd/otelcontribcol/go.mod:5]:
```suggestion
go 1.22.0

toolchain go1.22.7
```

**Review by codeboten — APPROVED** [review]
Will update the toolchain after this PR so we can get this merged

**Pull request merged**